src/modules/services-systemd/main.py

```diff
@@ -23,96 +23,61 @@
 
 
 def pretty_name():
-    return _("Configure systemd services")
+    return _("Configure systemd units")
 
 
-def systemctl(targets, command, suffix):
+def systemctl(units):
     """
-    For each entry in @p targets, run "systemctl <command> <thing>",
-    where <thing> is the entry's name plus the given @p suffix.
-    (No dot is added between name and suffix; suffix may be empty)
+    For each entry in @p units, run "systemctl <action> <name>",
+    where each unit is a map of unit name, action, and a flag.
 
     Returns a failure message, or None if this was successful.
-    Services that are not mandatory have their failures suppressed
+    Units that are not mandatory have their failures suppressed
     silently.
     """
-    for svc in targets:
-        if isinstance(svc, str):
-            name = svc
+
+    for unit in units:
+        if isinstance(unit, str):
+            name = unit
+            action = "enable"
             mandatory = False
         else:
-            name = svc["name"]
-            mandatory = svc.get("mandatory", False)
+            name = unit["name"]
```

Comment:
What happens here if `unit["name"]` is not populated.  It should probably write an error to the log and move onto the next service.

Reply:
Implemented. Although in that case (when the 'name' is missing), it violates the schema format which clearly specifies that the 'name' is required.